apps/api/src/deployment/services/deployment/deployment.service.ts

```diff
@@ -103,4 +110,41 @@ export class DeploymentService {
 
     return await this.findByOwnerAndDseq(wallet.address, dseq);
   }
+
+  public async update(wallet: UserWalletOutput, dseq: string, input: UpdateDeploymentRequest["data"]): Promise<GetDeploymentResponse["data"]> {
+    const { sdl, certificate } = input;
+
+    if (!this.sdlService.validateSdl(sdl)) {
+      throw new BadRequest("Invalid SDL");
+    }
+
+    const deployment = await this.findByOwnerAndDseq(wallet.address, dseq);
+    const manifestVersion = await this.sdlService.getManifestVersion(sdl, "beta3");
+    const manifest = this.sdlService.getManifest(sdl, "beta3", true) as string;
+
+    // If manifest version has changed, send update transaction
```

Comment:
non-blocking

the code would be cleaner and more readable if such things were extracted into dedicated methods. Comments wouldn't be needed. e.g.:

```typescript
async ensureDeploymentIsUpToDate(manifest, deployment) {
    if (Buffer.from(manifestVersion).toString("base64") !== deployment.deployment.version) {
      const message = this.rpcMessageService.getUpdateDeploymentMsg({
        owner: wallet.address,
        dseq,
        version: manifestVersion
      });

      await this.signerService.executeDecodedTxByUserId(wallet.userId, [message]);
    }
}
```